Description: Fixes error: couldn't allocate input reg for constraint 'vr'
 in source files generated by wasm2c.
Author: Nathan Pratta Teodosio <nteodosio@ubuntu.com>

--- a/media/libsoundtouch/src/moz.build
+++ b/media/libsoundtouch/src/moz.build
@@ -1,3 +1,5 @@
+if CONFIG['TARGET_CPU'] == 'riscv64':
+    DEFINES["WASM_RT_NONCONFORMING_ALLOW_OOB_READ_ELIMINATION"] = True
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
--- a/security/rlbox/rlbox.mozbuild
+++ b/security/rlbox/rlbox.mozbuild
@@ -1,3 +1,5 @@
+if CONFIG['TARGET_CPU'] == 'riscv64':
+    DEFINES["WASM_RT_NONCONFORMING_ALLOW_OOB_READ_ELIMINATION"] = True
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
